src/Protocols/Wishbone/Standard/Hedgehog.hs

```diff
@@ -25,63 +28,157 @@ import Protocols hiding (stallC, circuit)
 import Protocols.Hedgehog
 import Protocols.Wishbone
 import Prelude as P hiding (cycle)
+import GHC.Stack (HasCallStack)
 
 -- | Datatype representing a single transaction request sent from a Wishbone Master to a Wishbone Slave
 data WishboneMasterRequest addressWidth dat
-  = Read (BitVector addressWidth)
-  | Write (BitVector addressWidth) dat
-  deriving (Show)
+  = Read (BitVector addressWidth) (BitVector (BitSize dat `DivRU` 8))
+  | Write (BitVector addressWidth) (BitVector (BitSize dat `DivRU` 8)) dat
 
-data WishboneStandardState
-  = WSSQuiet
-  | WSSInCycleNoStrobe
-  | WSSWaitForSlave
+deriving instance (KnownNat addressWidth, KnownNat (BitSize a), Show a) => (Show (WishboneMasterRequest addressWidth a))
+
+--
+-- Validation for (lenient) spec compliance
+--
+-- The Wishbone spec mostly specifies the handshake protocols but does not make many
+-- assumptions about which signals *should* be valid in response to what. This is
+-- presumably done so to allow circuits to be very flexible, however it makes for
+-- a relatively weak validation tool.
+--
```

Comment:
This doesn't end up in Haddock, perhaps add it to some function or the module header?

Reply:
I added this to the module header!